### PR TITLE
fix: support override monitoring endpoint

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)
- [ ] Rollback plan is reviewed and LGTMed
- [ ] All new data plane features have a completed end to end testing plan

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java [hunk 1]
```diff
@@ -999,6 +1009,17 @@ public MetricsProvider getMetricsProvider() {
       return this.metricsProvider;
     }
 
+    /** Set the endpoint for publishing client side metrics. */
+    public Builder setMetricsEndpoint(String endpoint) {
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java:1013]: This needs a bit more documentation and be more explicit that this is for the builtin metrics

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java [hunk 2]
```diff
@@ -999,6 +1009,17 @@ public MetricsProvider getMetricsProvider() {
...
+      this.metricsEndpoint = endpoint;
+      return this;
+    }
+
+    /** Get the endpoint for client side metrics. */
+    public String getMetricsEndpoint() {
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java:1019]: please be more explicit that this is for built in metrics

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java [hunk 3]
```diff
@@ -126,14 +118,14 @@ public final class BigtableCloudMonitoringExporter implements MetricExporter {
...
-      String projectId, @Nullable Credentials credentials) throws IOException {
+      String projectId, @Nullable Credentials credentials, String endpoint) throws IOException {
     MetricServiceSettings.Builder settingsBuilder = MetricServiceSettings.newBuilder();
     CredentialsProvider credentialsProvider =
         Optional.ofNullable(credentials)
             .<CredentialsProvider>map(FixedCredentialsProvider::create)
             .orElse(NoCredentialsProvider.create());
     settingsBuilder.setCredentialsProvider(credentialsProvider);
-    settingsBuilder.setEndpoint(MONITORING_ENDPOINT);
+    settingsBuilder.setEndpoint(endpoint);
```
[the remark this answers is not in the document]
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java:128]: Currently we use it in our test environment to write to the test endpoint. I was debating if we want to keep it. Not sure if anyone ever discovered this and started using it.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsView.java [hunk 4]
```diff
@@ -37,19 +38,37 @@ private BuiltinMetricsView() {}
...
+  /**
+   * Register built-in metrics on the {@link SdkMeterProviderBuilder} with custom credentials and
+   * default endpoint.
+   */
   public static void registerBuiltinMetrics(
       String projectId, @Nullable Credentials credentials, SdkMeterProviderBuilder builder)
       throws IOException {
-    MetricExporter metricExporter = BigtableCloudMonitoringExporter.create(projectId, credentials);
+    BuiltinMetricsView.registerBuiltinMetrics(
+        projectId, credentials, builder, MetricServiceSettings.getDefaultEndpoint());
```
[the remark this answers is not in the document]
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsView.java:57]: This was the behavior we had in BigtableCloudMonitoringExporter. Whats the resolved universe endpoint?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java [hunk 5]
```diff
@@ -79,11 +78,11 @@ public final class BigtableCloudMonitoringExporter implements MetricExporter {
       Logger.getLogger(BigtableCloudMonitoringExporter.class.getName());
 
   // This system property can be used to override the monitoring endpoint
-  // to a different environment. It's meant for internal testing only.
+  // to a different environment. It's meant for internal testing only and
+  // will be removed in future versions. Use settings in EnhancedBigtableStubSettings
+  // to override the endpoint.
   private static final String MONITORING_ENDPOINT =
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java:84]: This is a bit confusing now - its no longer the monitoring endpoint, but a monitoring endpoint override from sys prop. How about `MONITORING_ENDPOINT_OVERRIDE_SYS_PROP`?

Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java:84]: Also mark it as Nullable and deprecated

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsView.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsView.java:57]: MetricServiceSettings.getDefaultEndpoint() - should this be null to flag default value?

**Review by igorbernstein2 — APPROVED** [review]

**Pull request merged**